### DISCUSSION TITLE
New db2 functions

### DISCUSSION THREAD

**Discussion by pabloto** [codefori/vscode-ibmi#588] [Ideas]
Hi, it will be wonderful to add two new functions.
The first one like run sql script in acs is the select for update.

The second is the possibility to copy a single row from a result set. After select a row from a result set, prepare a mask with the value from the rows selected and insert into table. 
Many thanks

**Comment by worksofliam** [comment]
@pabloto Thanks for these awesome ideas.

Is it possible you'd be able to share more details about these ideas? Any documentation to how the first idea works? I am not sure I've ever seen it.

**Reply by pabloto** [reply]
Hi, is quite simple, I don't know how it works, but if you try from run sql script
Select * from sample.employee for update

You can edit every field you want.

Many thanks

**Reply by worksofliam** [reply]
Wow, I have never seen that! That is super cool.

Sadly, I need a bunch of time to work on the new database tools, but I will keep this in mind for sure!!

**Reply by worksofliam** [reply]
Note to self: 

1. Find keys to table. If no keys, through error
2. Load data from select statement
3. Make all non-key fields editable with `contenteditable` 
4. When user presses enter or loses focus, update column using the row key(s)

**Reply by JonFP** [reply]
Technically even the key fields are editable in most cases.  Certainly those that were DDS created with designated key fields. Never tried it with DDL created tables.

On the subject of key fields - they don't seem to be identified in the Outline view.
